File: Top interview 150/0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        import heapq as hp

        # 모든 원소를 한번에 음수로 변환
        nums = [-n for n in nums]

        # heap으로 변환: O(n)
        hp.heapify(nums)

        # k번 pop: O(k log n)
        for i in range(k-1):
            hp.heappop(nums)
        
        return -hp.heappop(nums)

File: LeetCode 75/0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cs
public class Solution {
    public int FindKthLargest(int[] nums, int k) {
        // C#에서는 priority queue를 이용해서 표현
        PriorityQueue<int, int> pq = new PriorityQueue<int, int>();

        // 최대 힙처럼 작동하도록 음수로 설정
        // 시간 복잡도: O(n log n)
        foreach (int num in nums) {
            pq.Enqueue(num, -num);
        }

        // k-1번 pop 수행: O(k log n)
        for (int i = 0; i < k - 1; i++) {
            pq.Dequeue();
        }

        return pq.Dequeue();

    }
}